### PR TITLE
[no ticket][risk=no] e2e test framework general improvement

### PR DESCRIPTION
**Pull request by aweng98**
Rework `waitForText` func in `wait-utils.ts`, update usage.
Misc code fixes.

### CODE REVIEW

File: e2e/app/modal/workspace-review-research-purpose-modal.ts [hunk 1]
```diff
@@ -11,9 +11,13 @@ export default class WorkspaceReviewResearchPurposeModal extends Modal {
   }
 
   async isLoaded(): Promise<boolean> {
-    return await waitForText(this.page, modalTitle, { xpath: this.getXpath() }).catch(() => {
-      return false;
-    });
+    return await waitForText(this.page, modalTitle, { container: this })
+      .then(() => {
+        return true;
```
Comment (aweng98) [inline e2e/app/modal/workspace-review-research-purpose-modal.ts:16]: Missing `.then` in Promise chain.

File: e2e/app/container.ts [hunk 2]
```diff
@@ -31,10 +31,8 @@ export default class Container {
   }
 
   async waitUntilVisible(timeout = 60000): Promise<void> {
-    await Promise.all([
-      waitWhileLoading(this.page, timeout),
-      this.page.waitForXPath(this.getXpath(), { visible: true, timeout })
-    ]);
```
Comment (aweng98) [inline e2e/app/container.ts:37]: Fix a race condition bug: Loading spinner could come after `this.page.waitForXPath()` has finished.

File: e2e/app/modal/copy-to-workspace-modal.ts [hunk 3]
```diff
@@ -13,8 +13,18 @@ export default class CopyToWorkspaceModal extends Modal {
   }
 
   async isLoaded(): Promise<boolean> {
-    await waitForText(this.page, modalTitle, { xpath: this.getXpath() });
-    return true;
+    const timeout = 30 * 1000;
+    return Promise.all([
+      this.page.waitForXPath(`${this.getXpath()}//*[text()="${modalTitle}"]`, { visible: true, timeout }),
+      await this.getDestinationTextbox().exists(timeout),
+      await this.getNotebookNameTextbox().exists(timeout)
```
Comment (aweng98) [inline e2e/app/modal/copy-to-workspace-modal.ts:20]: Waits for modal textbox.

File: e2e/tests/nightly/user-access.spec.ts [hunk 4]
```diff
@@ -28,7 +28,7 @@ describe('User Access', () => {
 
     // the profile confirmation is expired, so the Review action is active for the profile
 
-    const reviewButton = await aarPage.getReviewProfileButton();
+    const reviewButton = aarPage.getReviewProfileButton();
```
Comment (aweng98) [inline e2e/tests/nightly/user-access.spec.ts:31]: Non-async func. Don't need `await`.

File: e2e/app/modal/export-to-notebook-modal.ts [hunk 5]
```diff
@@ -44,7 +44,7 @@ export default class ExportToNotebookModal extends Modal {
     return radio.select();
   }
 
-  async pickAnalysisTool(analysisTool: AnalysisTool = AnalysisTool.Hail): Promise<RadioButton> {
+  pickAnalysisTool(analysisTool: AnalysisTool = AnalysisTool.Hail): RadioButton {
```
Comment (aweng98) [inline e2e/app/modal/export-to-notebook-modal.ts:47]: Non-async func.

**Review by yonghaoy — APPROVED** [review]
LGTM

**Pull request merged**